src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/AssemblyContigAlignmentSignatureClassifier.java

```diff
@@ -0,0 +1,171 @@
+                }
+        );
+
+        // split between the case where both alignments has unique ref span or not
+        final Tuple2<JavaRDD<AlignedContig>, JavaRDD<AlignedContig>> hasFullyContainedRefSpanOrNot =
+                RDDUtils.split(preprocessedContigs, AssemblyContigAlignmentSignatureClassifier::hasIncompletePictureDueToRefSpanContainment, false);
+        contigsByRawTypes.put(RawTypes.Incomplete, hasFullyContainedRefSpanOrNot._1);
+
+        // split between same chromosome mapping or not
+        final Tuple2<JavaRDD<AlignedContig>, JavaRDD<AlignedContig>> sameChrOrNot =
+                RDDUtils.split(hasFullyContainedRefSpanOrNot._2, AssemblyContigAlignmentSignatureClassifier::isSameChromosomeMapping, false);
+        contigsByRawTypes.put(RawTypes.TandemDupOrMEIBkpt, sameChrOrNot._2);
+
+        // split between strand switch or not (NOTE BOTH SAME CHROMOSOME MAPPING)
+        final Tuple2<JavaRDD<AlignedContig>, JavaRDD<AlignedContig>> strandSwitchOrNot =
+                RDDUtils.split(sameChrOrNot._1, AssemblyContigAlignmentSignatureClassifier::indicatesIntraChrStrandSwitchBkpts, false);
+        contigsByRawTypes.put(RawTypes.IntraChrStrandSwitch, strandSwitchOrNot._1);
+
+        // split between ref block switch or not (NOTE BOTH SAME CHROMOSOME MAPPING AND NO STRAND SWITCH)
+        final Tuple2<JavaRDD<AlignedContig>, JavaRDD<AlignedContig>> tandemDupBkptOrSimpleInsDel =
+                RDDUtils.split(strandSwitchOrNot._2, AssemblyContigAlignmentSignatureClassifier::indicatesIntraChrTandemDupBkpts, false);
+        contigsByRawTypes.put(RawTypes.InsDel, tandemDupBkptOrSimpleInsDel._2);
+        contigsByRawTypes.put(RawTypes.TandemDupOrMEIBkpt, tandemDupBkptOrSimpleInsDel._1);
+    }
+
+    //==================================================================================================================
+
+    // TODO: 10/27/17 add tests for these predicates
+
+    @VisibleForTesting
+    static boolean hasIncompletePictureDueToRefSpanContainment(final AlignedContig contigWithOnlyOneConfigAnd2Aln) {
+        Utils.validateArg(hasOnly2Alignments(contigWithOnlyOneConfigAnd2Aln),
+                "assumption that input contig has only 2 alignments is violated. \n" +
+                        contigWithOnlyOneConfigAnd2Aln.toString());
+        final AlignmentInterval one = contigWithOnlyOneConfigAnd2Aln.alignmentIntervals.get(0),
+                                two = contigWithOnlyOneConfigAnd2Aln.alignmentIntervals.get(1);
+        return one.referenceSpan.contains(two.referenceSpan)
+                ||
+                two.referenceSpan.contains(one.referenceSpan);
+    }
+
+    /**
+     * @param contigWithOnlyOneConfigAnd2Aln assumed to have only two alignments
+     */
+    @VisibleForTesting
+    static boolean isSameChromosomeMapping(final AlignedContig contigWithOnlyOneConfigAnd2Aln) {
+        Utils.validateArg(!hasIncompletePictureDueToRefSpanContainment(contigWithOnlyOneConfigAnd2Aln),
```

Comment:
This assumption seems weird given the method name and what it does. I'd remove it.

Reply:
done